### PR TITLE
PromQL: Define and document behavior of quantile and histogram_quantile with φ=NaN

### PR DESCRIPTION
**Pull request by jyz0309**
Resolves https://github.com/prometheus/prometheus/issues/10117

### CODE REVIEW

File: promql/quantile.go [hunk 1]
```diff
@@ -181,11 +186,12 @@ func ensureMonotonic(buckets buckets) {
 // quantile calculates the given quantile of a vector of samples.
 //
 // The Vector will be sorted.
-// If 'values' has zero elements, NaN is returned.
+// If 'values' has zero elements
+// If q==NaN, NaN is returned.
```
Comment (beorn7) [inline promql/quantile.go:190]: An "or" is missing, and the 2nd "If" needs to be lower-case. (Or Add ", NaN is returned." at the end of line 189.)

File: docs/querying/operators.md [hunk 2]
```diff
@@ -241,7 +241,7 @@ vector. `by` and `without` are only used to bucket the input vector.
 `quantile` calculates the φ-quantile, the value that ranks at number φ*N among
 the N metric values of the dimensions aggregated over. φ is provided as the
 aggregation parameter. For example, `quantile(0.5, ...)` calculates the median,
-`quantile(0.95, ...)` the 95th percentile.
+`quantile(0.95, ...)` the 95th percentile. For φ = `NaN`, `NaN` is returned.
```
Comment (beorn7) [inline docs/querying/operators.md:244]: Now that you are on it, could you also add the "For φ < 0, `-Inf` is returned. For φ > 1, `+Inf` is returned." that we already had for `histogram_quantile` above.

File: promql/testdata/histograms.test [hunk 3]
```diff
@@ -60,6 +60,11 @@ eval instant at 50m histogram_quantile(1.01, testhistogram_bucket)
 	{start="positive"} +Inf
 	{start="negative"} +Inf
 
+# Quantile invalid.
+eval instant at 50m histogram_quantile(NaN, testhistogram_bucket)
+	{start="positive"} NaN
+	{start="negative"} NaN
```
Comment (beorn7) [inline promql/testdata/histograms.test:66]: Could you add the same test for `quantile(NaN, ...)`?

**Review by beorn7 — CHANGES_REQUESTED** [review]
Looks good in general, thank you very much.

Just a few doc nits, and a request for more tests.

**Review by beorn7 — APPROVED** [review]
Looks good, thank you.

Let's give @roidelapluie or @codesome a moment to chime in before merging this.

**Review by roidelapluie — APPROVED** [review]

**Comment by beorn7** [comment]
Closing and opening to kick CircleCI.

**Pull request merged**